src/nbd.c

```diff
@@ -869,8 +874,14 @@ gboolean r_nbd_run_server(gint sock, GError **error)
 				xfer->reply.error = GUINT32_TO_BE(5); /* NBD_EIO */
 				xfer->done = TRUE;
 			} else {
-				g_message("request failed (retrying)");
-				xfer->errors++;
+				if (response_code == 404) {
+					g_message("request failed (not found)");
+					xfer->reply.error = GUINT32_TO_BE(5); /* NBD_EIO */
+					xfer->done = TRUE;
+				} else {
+					g_message("request failed (retrying)");
+					xfer->errors++;
+				}
```

Comment:
Couldn't this be just another `else if`, instead of being nested in the `else`?

Reply:
Yes, I had this in mind but wasn't sure which variant to choose. Will change.